medicines/doc-index-updater/src/storage_client/sftp_client.rs

```diff
@@ -1,35 +1,30 @@
-use crate::{
-    models::FileSource,
-    storage_client::{
-        models::{StorageClientError, StorageFile},
-        AzureBlobStorage, StorageClient,
-    },
+use super::{
+    models::{BlobResponse, SftpError, StorageClientError},
+    GetBlob,
 };
 use anyhow::anyhow;
 use async_ssh2::{Session, Sftp};
+use async_trait::async_trait;
 use std::net::TcpStream;
-use thiserror::Error;
 use tokio::io::AsyncReadExt;
 
-#[derive(Error, Debug)]
-pub enum SftpError {
-    #[error("A TCP error connecting to server. ({0:?})")]
-    TcpError(#[from] std::io::Error),
-    #[error("An SSH error connecting to server. ({0:?})")]
-    Ssh2Error(#[from] async_ssh2::Error),
-    #[error("File could not be retrieved on server")]
-    CouldNotRetrieveFile,
-    #[error(transparent)]
-    Other(#[from] anyhow::Error),
+struct SftpInfo {
```

Comment:
Maybe "SftpConfig" would be more descriptive here? I think that would make it's use within `SftpClient` clearer, IMO.

Reply:
agreed